As a user, I don't expect index out of bounds exceptions when I try to evaluation 10 days of NWM AnA with sampling uncertainty

The declaration is below. The exception: 

```
wres.pipeline.InternalWresException: Could not complete project execution
	at wres.pipeline.Evaluator.evaluate(Evaluator.java:407)
	at wres.pipeline.Evaluator.evaluate(Evaluator.java:179)
	at wres.Functions.evaluate(Functions.java:135)
	at wres.server.EvaluationService.lambda$startEvaluation$2(EvaluationService.java:695)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: wres.pipeline.WresProcessingException: Encountered an error while processing evaluation 'XMck6KYqPwl4P7NslRAAZkTTQfA': 
	at wres.pipeline.Evaluator.evaluate(Evaluator.java:842)
	at wres.pipeline.Evaluator.evaluate(Evaluator.java:378)
	... 7 common frames omitted
Caused by: java.util.concurrent.CompletionException: wres.datamodel.bootstrap.ResamplingException: Encountered an error while attempting to resample a time-series.
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770)
	... 3 common frames omitted
Caused by: wres.datamodel.bootstrap.ResamplingException: Encountered an error while attempting to resample a time-series.
	at wres.datamodel.bootstrap.StationaryBootstrapResampler.resample(StationaryBootstrapResampler.java:720)
	at wres.datamodel.bootstrap.StationaryBootstrapResampler.resample(StationaryBootstrapResampler.java:159)
	at wres.pipeline.pooling.PoolProcessor.getSamplingUncertaintyStatistics(PoolProcessor.java:597)
	at wres.pipeline.pooling.PoolProcessor.createStatistics(PoolProcessor.java:492)
	at wres.pipeline.pooling.PoolProcessor.get(PoolProcessor.java:151)
	at wres.pipeline.pooling.PoolProcessor.get(PoolProcessor.java:60)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	... 3 common frames omitted
Caused by: java.util.concurrent.ExecutionException: java.lang.IndexOutOfBoundsException: Index: 1 Size: 1
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
	at wres.datamodel.bootstrap.StationaryBootstrapResampler.resample(StationaryBootstrapResampler.java:708)
	... 9 common frames omitted
Caused by: java.lang.IndexOutOfBoundsException: Index: 1 Size: 1
	at java.base/java.util.ImmutableCollections$AbstractImmutableList.outOfBounds(ImmutableCollections.java:333)
	at java.base/java.util.ImmutableCollections$List12.get(ImmutableCollections.java:585)
	at wres.datamodel.bootstrap.StationaryBootstrapResampler.lambda$getTimeSeriesResampler$1(StationaryBootstrapResampler.java:760)
	at wres.datamodel.bootstrap.StationaryBootstrapResampler.lambda$resample$0(StationaryBootstrapResampler.java:698)
	... 4 common frames omitted
```

I'll attempt to gather some data to reproduce, but it likely won't be easy.  I'll see if it can be reproduced using WRDS; that would be easier.

Thanks,

Hank

==============

```yaml
label: ISED Routine Evaluation Draft 1 AnA
observed:
  label: USGS NWIS Streamflow Observations
  sources:
  - interface: usgs nwis
    uri: https://nwis.waterservices.usgs.gov/nwis/iv
  variable:
    name: '00060'
predicted:
  label: NWC DStore Unassimilated AnA
  sources:
  - interface: nwm analysis assim no da channel rt conus
    uri: https://DStore_omitted/nwm/3.0/
  variable:
    name: streamflow
unit: ft3/s
valid_dates:
  minimum: '2025-02-11T00:00:00Z'
  maximum: '2025-02-21T16:19:20Z'
features:
- observed: '01631000'
  predicted: '5907079'
metrics:
- name: bias fraction
sampling_uncertainty:
  sample_size: 1000
  quantiles:
  - 0.05
  - 0.95
output_formats:
- format: csv2
- format: png
- format: netcdf
```

After adding the correct directory structure for the data, reproduced:

```
Caused by: java.util.concurrent.CompletionException: wres.datamodel.bootstrap.ResamplingException: Encountered an error while attempting to resample a time-series.
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: wres.datamodel.bootstrap.ResamplingException: Encountered an error while attempting to resample a time-series.
	at wres.datamodel.bootstrap.StationaryBootstrapResampler.resample(StationaryBootstrapResampler.java:720)
	at wres.datamodel.bootstrap.StationaryBootstrapResampler.resample(StationaryBootstrapResampler.java:159)
	at wres.pipeline.pooling.PoolProcessor.getSamplingUncertaintyStatistics(PoolProcessor.java:613)
	at wres.pipeline.pooling.PoolProcessor.createStatistics(PoolProcessor.java:508)
	at wres.pipeline.pooling.PoolProcessor.get(PoolProcessor.java:152)
	at wres.pipeline.pooling.PoolProcessor.get(PoolProcessor.java:61)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	... 3 common frames omitted
Caused by: java.util.concurrent.ExecutionException: java.lang.IndexOutOfBoundsException: Index: 1 Size: 1
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
	at wres.datamodel.bootstrap.StationaryBootstrapResampler.resample(StationaryBootstrapResampler.java:708)
	... 9 common frames omitted
Caused by: java.lang.IndexOutOfBoundsException: Index: 1 Size: 1
	at java.base/java.util.ImmutableCollections$AbstractImmutableList.outOfBounds(ImmutableCollections.java:333)
	at java.base/java.util.ImmutableCollections$List12.get(ImmutableCollections.java:585)
	at wres.datamodel.bootstrap.StationaryBootstrapResampler.lambda$getTimeSeriesResampler$1(StationaryBootstrapResampler.java:760)
	at wres.datamodel.bootstrap.StationaryBootstrapResampler.lambda$resample$0(StationaryBootstrapResampler.java:698)
	... 4 common frames omitted
```

More next week.

--- a/src/wres/pipeline/Evaluator.java
+++ b/src/wres/pipeline/Evaluator.java
@@ -682,11 +682,13 @@ public class Evaluator
                                                            .map( FeatureTuple::getGeometryTuple )
                                                            .collect( Collectors.toUnmodifiableSet() );
             Set<FeatureGroup> featureGroups = new TreeSet<>( project.getFeatureGroups() );
+
             Set<FeatureGroup> adjustedFeatureGroups =
                     EvaluationUtilities.adjustFeatureGroupsForSummaryStatistics( featureGroups,
                                                                                  unwrappedFeatures,
                                                                                  declaration.summaryStatistics(),
                                                                                  doNotPublish );
+
             EvaluationUtilities.createNetcdfBlobs( netcdfWriters,
                                                    adjustedFeatureGroups,
                                                    metricsAndThresholds );

--- a/wres-config/src/wres/config/yaml/DeclarationInterpolator.java
+++ b/wres-config/src/wres/config/yaml/DeclarationInterpolator.java
@@ -2452,8 +2452,13 @@ public class DeclarationInterpolator
 
         DataType calculatedDataType;
 
-        // Analysis durations present? If so, assume analyses
-        if ( DeclarationUtilities.hasAnalysisTimes( builder ) )
+        // Analysis durations present? If so, assume analyses, unless there is an analysis interface declared for
+        // another side of data
+        if ( DeclarationUtilities.hasAnalysisTimes( builder )
+             && DeclarationInterpolator.noAnalysisInterface( builder.right() )
+             && ( !DeclarationUtilities.hasBaseline( builder )
+                  || DeclarationInterpolator.noAnalysisInterface( builder.baseline()
+                                                                         .dataset() ) ) )
         {
             calculatedDataType = DataType.ANALYSES;
 
@@ -2496,7 +2501,8 @@ public class DeclarationInterpolator
             // Is it consistent with the type inferred from the declaration? If not, we only emit an error if the
             // type inferred from the declaration is ANALYSES because this requires definitive/unique declaration
             // options. Otherwise, we emit a warning.
-            if ( ingestedDataType != calculatedDataType && calculatedDataType == DataType.ANALYSES )
+            if ( ingestedDataType != calculatedDataType
+                 && calculatedDataType == DataType.ANALYSES )
             {
                 EvaluationStatusEvent event
                         = EvaluationStatusEvent.newBuilder()
@@ -2504,7 +2510,7 @@ public class DeclarationInterpolator
                                                .setEventMessage( THE_DATA_TYPE_INFERRED_FROM_THE_TIME_SERIES_DATA
                                                                  + "for "
                                                                  + article
-                                                                 + "'"
+                                                                 + " '"
                                                                  + orientation
                                                                  + DATASET_WAS
                                                                  + ingestedDataType
@@ -2552,6 +2558,25 @@ public class DeclarationInterpolator
         datasetBuilder.type( typeToUse );
 
         return Collections.unmodifiableList( events );
+    }
+
+    /**
+     * @param dataset the dataset
+     * @return whether the dataset has an analysis interface
+     */
+
+    private static boolean noAnalysisInterface( Dataset dataset )
+    {
+        if ( Objects.isNull( dataset ) )
+        {
+            return true;
+        }
+
+        return dataset.sources()
+                      .stream()
+                      .noneMatch( s -> Objects.nonNull( s.sourceInterface() )
+                                       && s.sourceInterface()
+                                           .isAnalysisInterface() );
     }
 
     /**

--- a/wres-config/src/wres/config/yaml/components/SourceInterface.java
+++ b/wres-config/src/wres/config/yaml/components/SourceInterface.java
@@ -92,19 +92,20 @@ public enum SourceInterface
     NWM_LONG_RANGE_CHANNEL_RT_CONUS( Set.of( DataType.SINGLE_VALUED_FORECASTS ), FeatureAuthority.NWM_FEATURE_ID ),
     /** nwm short range channel rt alaska. */
     @JsonProperty( "nwm short range channel rt alaska" )
-    NWM_SHORT_RANGE_CHANNEL_RT_CONUS_ALASKA( Set.of( DataType.SINGLE_VALUED_FORECASTS ), FeatureAuthority.NWM_FEATURE_ID ),
+    NWM_SHORT_RANGE_CHANNEL_RT_CONUS_ALASKA( Set.of( DataType.SINGLE_VALUED_FORECASTS ),
+                                             FeatureAuthority.NWM_FEATURE_ID ),
     /** nwm medium range ensemble channel rt alaska. */
     @JsonProperty( "nwm medium range ensemble channel rt alaska" )
     NWM_MEDIUM_RANGE_ENSEMBLE_CHANNEL_RT_ALASKA( Set.of( DataType.ENSEMBLE_FORECASTS ),
-                                                FeatureAuthority.NWM_FEATURE_ID ),
+                                                 FeatureAuthority.NWM_FEATURE_ID ),
     /** nwm medium range deterministic channel rt alaska. */
     @JsonProperty( "nwm medium range deterministic channel rt alaska" )
     NWM_MEDIUM_RANGE_DETERMINISTIC_CHANNEL_RT_ALASKA( Set.of( DataType.SINGLE_VALUED_FORECASTS ),
-                                                     FeatureAuthority.NWM_FEATURE_ID ),
+                                                      FeatureAuthority.NWM_FEATURE_ID ),
     /** nwm medium range no da deterministic channel rt alaska. */
     @JsonProperty( "nwm medium range no da deterministic channel rt alaska" )
     NWM_MEDIUM_RANGE_NO_DA_DETERMINISTIC_CHANNEL_RT_ALASKA( Set.of( DataType.SINGLE_VALUED_FORECASTS ),
-                                                           FeatureAuthority.NWM_FEATURE_ID ),
+                                                            FeatureAuthority.NWM_FEATURE_ID ),
     /** nwm analysis assim channel rt alaska. */
     @JsonProperty( "nwm analysis assim channel rt alaska" )
     NWM_ANALYSIS_ASSIM_CHANNEL_RT_ALASKA( Set.of( DataType.ANALYSES ), FeatureAuthority.NWM_FEATURE_ID ),
@@ -168,5 +169,18 @@ public enum SourceInterface
     {
         return this.name()
                    .startsWith( "NWM_" );
+    }
+
+    /**
+     * Convenience method that inspects the interface {@link #name()} and returns <code>true</code> when the name
+     * contains 'ANALYSIS_', otherwise <code>false</code>.
+     *
+     * @return whether the source interface is an analysis interface
+     */
+
+    public boolean isAnalysisInterface()
+    {
+        return this.name()
+                   .contains( "ANALYSIS_" );
     }
 }

--- a/wres-datamodel/src/wres/datamodel/bootstrap/BootstrapPool.java
+++ b/wres-datamodel/src/wres/datamodel/bootstrap/BootstrapPool.java
@@ -89,7 +89,7 @@ class BootstrapPool<T>
     /**
      * Returns the time-series with all events present
      *
-     * @return the time-series with at least the number if requested events
+     * @return the time-series with all events
      */
 
     List<List<Event<T>>> getTimeSeriesWithAllEvents()

--- a/wres-datamodel/src/wres/datamodel/bootstrap/StationaryBootstrapResampler.java
+++ b/wres-datamodel/src/wres/datamodel/bootstrap/StationaryBootstrapResampler.java
@@ -198,7 +198,7 @@ public class StationaryBootstrapResampler<T>
         // ordering
         List<List<TimeSeries<T>>> groupedBySize = pool.getOrderedTimeSeries();
 
-        // One ResampledIndexes for each time-series, indicating where to obtain the event values for that series
+        // One ResampleIndexes for each time-series, indicating where to obtain the event values for that series
         List<ResampleIndexes> outerIndexes = new ArrayList<>();
         for ( List<TimeSeries<T>> poolSeries : groupedBySize )
         {
@@ -211,8 +211,7 @@ public class StationaryBootstrapResampler<T>
 
                 // Forecast time-series which are probably non-stationary across lead durations, unless they are based
                 // on climatology
-                if ( !nextSeries.getReferenceTimes()
-                                .isEmpty() )
+                if ( pool.hasForecasts() )
                 {
                     nextIndexes = this.generateResampleIndexesForForecastSeries( nextSeries,
                                                                                  i,
@@ -756,6 +755,29 @@ public class StationaryBootstrapResampler<T>
                 Event<T> nextEvent = events.get( j );
                 int[] index = indexes.indexes()
                                      .get( j );
+
+                if ( eventsToSample.size() <= index[0] )
+                {
+                    throw new IndexOutOfBoundsException( "While attempting to resample a time-series at index "
+                                                         + index[0]
+                                                         + ", discovered a maximum time-series index of "
+                                                         + ( eventsToSample.size() - 1 )
+                                                         + ", which is smaller than the required index." );
+                }
+
+                if ( eventsToSample.get( index[0] )
+                                   .size() <= index[1] )
+                {
+                    throw new IndexOutOfBoundsException( "While attempting to resample a time-series event at index "
+                                                         + index[1]
+                                                         + " of the time-series at index "
+                                                         + index[0]
+                                                         + ", discovered a maximum time-series event index of "
+                                                         + ( eventsToSample.get( index[0] )
+                                                                           .size() - 1 )
+                                                         + ", which is smaller than the required index." );
+                }
+
                 Event<T> resampledValue = eventsToSample.get( index[0] )
                                                         .get( index[1] );
                 Event<T> resampled = Event.of( nextEvent.getTime(), resampledValue.getValue() );

--- a/wres-datamodel/src/wres/datamodel/time/TimeSeriesSlicer.java
+++ b/wres-datamodel/src/wres/datamodel/time/TimeSeriesSlicer.java
@@ -1384,7 +1384,8 @@ public final class TimeSeriesSlicer
         Objects.requireNonNull( referenceTimeTypes );
 
         return referenceTimeTypes.stream()
-                                 .anyMatch( t -> t == ReferenceTimeType.T0 || t == ReferenceTimeType.ISSUED_TIME );
+                                 .anyMatch( t -> t == ReferenceTimeType.T0
+                                                 || t == ReferenceTimeType.ISSUED_TIME );
     }
 
     /**

--- a/wres-io/src/wres/io/retrieving/database/AnalysisRetriever.java
+++ b/wres-io/src/wres/io/retrieving/database/AnalysisRetriever.java
@@ -16,16 +16,13 @@ import wres.io.retrieving.DuplicatePolicy;
 import wres.io.retrieving.RetrieverUtilities;
 
 /**
- * <p>Retrieves data from the wres.TimeSeries and wres.TimeSeriesValue tables but
- * in the pattern expected for treating the nth timestep of each analysis as if
- * it were an event in a timeseries across analyses, sort of like observations.
+ * <p>Retrieves data from the wres.TimeSeries and wres.TimeSeriesValue tables but in the pattern expected for treating
+ * the nth timestep of each analysis as if it were an event in a time-series across analyses, sort of like observations.
  *
- * <p>The reason for separating it from forecast and observation timeseries
- * retrieval is that each analysis has N events in an actual timeseries, but the
- * structure and use of the analyses and origin of analyses differs from both
- * observation and timeseries. The structure of an NWM analysis, for example, is
- * akin to an NWM forecast, with a reference datetime and valid datetimes.
- * However, when using the analyses in an evaluation of forecasts, one event
+ * <p>The reason for separating it from forecast and observation timeseries retrieval is that each analysis has N
+ * events in an actual timeseries, but the structure and use of the analyses and origin of analyses differs from both
+ * observation and timeseries. The structure of an NWM analysis, for example, is akin to an NWM forecast, with a
+ * reference datetime and valid datetimes. However, when using the analyses in an evaluation of forecasts, one event
  * from each analysis is picked out and a broader timeseries is created.
  */
 
@@ -103,7 +100,7 @@ class AnalysisRetriever extends TimeSeriesRetriever<Double>
 
         /**
          * Sets the earliest analysis hour, if not <code>null</null>.
-         * 
+         *
          * @param earliestAnalysisDuration duration
          * @return A builder
          */
@@ -119,7 +116,7 @@ class AnalysisRetriever extends TimeSeriesRetriever<Double>
 
         /**
          * Set the latest analysis hour, if not <code>null</null>.
-         * 
+         *
          * @param latestAnalysisDuration duration
          * @return A builder
          */
@@ -173,7 +170,7 @@ class AnalysisRetriever extends TimeSeriesRetriever<Double>
 
     /**
      * Returns the earliest analysis duration or null.
-     * 
+     *
      * @return the earliest analysis duration or null
      */
 
@@ -184,7 +181,7 @@ class AnalysisRetriever extends TimeSeriesRetriever<Double>
 
     /**
      * Returns the latest analysis duration or null.
-     * 
+     *
      * @return the latest analysis duration or null
      */
 

--- a/wres-reading/src/wres/reading/wrds/thresholds/WrdsThresholdReader.java
+++ b/wres-reading/src/wres/reading/wrds/thresholds/WrdsThresholdReader.java
@@ -720,7 +720,7 @@ public class WrdsThresholdReader implements ThresholdReader
                          "reconciled with features to evaluate. Features without ",
                          "thresholds will be skipped. If the number of features ",
                          "without thresholds is larger than expected, ensure that ",
-                         "the source of feature names (featureNameFrom) is properly ",
+                         "the source of feature names (feature_name_from) is properly ",
                          "declared for the external thresholds. The ",
                          "declared features without thresholds are: ",
                          featureNamesWithoutThresholds,

